apm-agent-plugins/apm-jdbc-plugin/src/test/java/co/elastic/apm/agent/jdbc/helper/ConnectionMetaDataTest.java

```diff
@@ -33,297 +35,333 @@ class ConnectionMetaDataTest {
-        testUrl("jjdbc:derby:memory:mydb", "derby", "localhost", -1);
-        testUrl("jjdbc:derby:mydb", "derby", "localhost", -1);
+        // https://db.apache.org/derby/docs/10.5/devguide/rdevdvlp22102.html
+        testUrl("jdbc:derby://my.host/memory:mydb;prop1=val1;prop2=val2", "derby", "my.host", 1527, "mydb");
+        testUrl("jdbc:derby://my.host/memory:mydb;prop1=val1;prop2=val2", "derby", "my.host", 1527, "mydb");
+        testUrl("jdbc:derby://my.host:666/memory:mydb;prop1=val1;prop2=val2", "derby", "my.host", 666, "mydb");
+        testUrl("jdbc:derby:jar:/mydb;prop1=val1;prop2=val2", "derby", "localhost", -1, "/mydb");
+        testUrl("jdbc:derby:memory:mydb", "derby", "localhost", -1, "mydb");
+        testUrl("jdbc:derby:mydb", "derby", "localhost", -1, "mydb");
+        testUrl("jdbc:derby:classpath:/myDB", "derby", "localhost", -1, "/myDB");
+        testUrl("jdbc:derby:jar:(C:/dbs.jar)products/boiledfood", "derby", "localhost", -1, "(C:/dbs.jar)products/boiledfood");
+        testUrl("jdbc:derby:directory:myDB", "derby", "localhost", -1, "myDB");
     }
 
     @Test
     void testHsqldb() {
         // http://hsqldb.org/doc/2.0/guide/dbproperties-chapt.html
-        testUrl("jdbc:hsqldb:file:~/mydb", "hsqldb", "localhost", -1);
-        testUrl("jdbc:hsqldb:file:enrolments;user=aUserName;ifexists=true", "hsqldb", "localhost", -1);
-        testUrl("jdbc:hsqldb:hsql://localhost/enrolments;close_result=true", "hsqldb", "localhost", 9001);
-        testUrl("jdbc:hsqldb:hsql://my.host/enrolments;close_result=true", "hsqldb", "my.host", 9001);
-        testUrl("jdbc:hsqldb:http://192.0.0.10:9500", "hsqldb", "192.0.0.10", 9500);
-        testUrl("jdbc:hsqldb:http://dbserver.somedomain.com", "hsqldb", "dbserver.somedomain.com", 9001);
-        testUrl("jdbc:hsqldb:mem:", "hsqldb", "localhost", -1);
+        testUrl("jdbc:hsqldb:file:~/mydb", "hsqldb", "localhost", -1, "~/mydb");
+        testUrl("jdbc:hsqldb:file:enrolments;user=aUserName;ifexists=true", "hsqldb", "localhost", -1, "enrolments");
+        testUrl("jdbc:hsqldb:hsql://localhost/enrolments;close_result=true", "hsqldb", "localhost", 9001, "enrolments");
+        testUrl("jdbc:hsqldb:hsql://my.host/enrolments;close_result=true", "hsqldb", "my.host", 9001, "enrolments");
+        testUrl("jdbc:hsqldb:http://192.0.0.10:9500", "hsqldb", "192.0.0.10", 9500, null);
+        testUrl("jdbc:hsqldb:http://dbserver.somedomain.com", "hsqldb", "dbserver.somedomain.com", 9001, null);
+        testUrl("jdbc:hsqldb:mem:", "hsqldb", "localhost", -1, null);
+        testUrl("jdbc:hsqldb:res:/adirectory/dbname", "hsqldb", "localhost", -1, "/adirectory/dbname");
     }
 
     @Test
     void testInvalid() {
-        testUrl("postgresql://myhost:666/database", "unknown", null, -1);
+        // invalid URL, just checking the defaults, assuming "localhost" and no port as if it would be an in-memory database
+        testUrl("", "unknown", "localhost", -1, null);
```

Comment:
```suggestion
        testUrl("", "unknown", "localhost", -1, null);
        testUrl("wrong", "unknown", "localhost", -1, "wrong");
```

Reply:
The fact that we have `wrong` as a parsed database name is not intentional, but the implementation makes it work this way.
I will leave it as-is for now as it is really a corner case, and we can revisit later if required.